store/displayPreferencesApi.ts

```diff
@@ -0,0 +1,216 @@
+ * @returns {ClientPreferences} - The response with the correct datatypes.
+ */
+function castResponse(data: DisplayPreferencesDto): ClientPreferences {
+  const result = data as ClientPreferences;
+  for (const [key, value] of Object.entries(
+    result.CustomPrefs as CustomPreferences
+  )) {
+    /**
+     * destr does proper conversion for all the types, even undefined
+     */
+    // @ts-expect-error - TypeScript can't infer types from Object.entries
+    result.CustomPrefs?.[key] = destr(value);
+  }
+  return result;
+}
+
+export const state = defaultState;
+
+export const mutations: MutationTree<DisplayPreferencesApiState> = {
+  RESET_STATE(state: DisplayPreferencesApiState) {
+    Object.assign(state, defaultState);
+  },
+  SYNCING_STARTED(state: DisplayPreferencesApiState) {
+    state.syncing = true;
+  },
+  SYNCING_ENDED(state: DisplayPreferencesApiState) {
+    state.syncing = false;
+    state.LastSync = Date.now();
+  },
+  UPDATE_CLIENT_SETTINGS(state: DisplayPreferencesApiState) {
+    state.LastSettingChange = Date.now();
+  }
+};
+
+export const actions: ActionTree<
+  DisplayPreferencesApiState,
+  DisplayPreferencesApiState
+> = {
+  /**
+   * Fetches display preferences and stores them at boot time
+   *
+   * @param {any} context - Vuex action context
+   * @param {any} context.dispatch - Vuex dispatch
+   */
+  async initState({ commit, dispatch }) {
+    if (this.$auth.loggedIn) {
+      commit('SYNCING_STARTED');
+      try {
+        const response = await this.$api.displayPreferences.getDisplayPreferences(
+          {
+            displayPreferencesId: 'usersettings',
+            userId: this.$auth.user?.Id,
+            client: 'vue'
+          }
+        );
+
+        if (response.status !== 200) {
+          throw new Error(
+            'get display preferences status response = ' + response.status
+          );
+        }
+
+        const data = castResponse(response.data);
+        if (data.CustomPrefs) {
+          /**
+           * We delete the keys that are not defined in the state's default state, so removed
+           * values locally don't pollute server's displayPreferences.
+           */
+          for (const key of Object.keys(data.CustomPrefs)) {
+            if (!(key in settingState())) {
+              // @ts-expect-error - TypeScript can't infer indexes typings from Object.keys
+              delete data.CustomPrefs[key];
+            }
+          }
+          commit(
+            'settings/INIT_STATE',
+            { data: data.CustomPrefs },
+            { root: true }
+          );
+        }
+      } catch (error) {
+        const message = this.$i18n.t('failedRetrievingDisplayPreferences');
+        dispatch(
+          'snackbar/pushSnackbarMessage',
+          {
+            message,
+            color: 'error'
+          },
+          { root: true }
+        );
+      }
+      commit('SYNCING_ENDED');
+    }
+  },
+  /**
+   * Pushes the current state to the server
```

Comment:
This seems a bit generic for a description. What's in this state? :yum: 

```suggestion
   * Pushes the client's current display preferences to the server
```